3D-Engine/Shader.cpp

```diff
@@ -0,0 +1,120 @@
+#include <iostream>
+#include <fstream>
+#include "Shader.h"
+
+
+Shader::Shader(const char* vertexPath, const char* fragmentPath)
+{
+    std::string VertexSource = Load(vertexPath);
+    std::string FragmentSource = Load(fragmentPath);
+
+    GLuint VertexID = Compile(GL_VERTEX_SHADER, &VertexSource);
+    GLuint FragmentID = Compile(GL_FRAGMENT_SHADER, &FragmentSource);
+
+    m_ProgramID = glCreateProgram();
+    glAttachShader(m_ProgramID, VertexID);
+    glAttachShader(m_ProgramID, FragmentID);
+    glLinkProgram(m_ProgramID);
+
+    LinkError();
+}
+
+std::string Shader::Load(const char* path)
+{
+    std::ifstream file;
+    std::string source, line;
+
+    file.open(path, std::ios::in);
+
+    if (!file.is_open())
+    {
+        std::cerr << std::string(path) + std::string(" failed to open") << std::endl;
+        exit(-1);
+    }
+
+    while (std::getline(file, line))
+    {
+        source += line;
+        source += '\n';
+    }
```

Comment:
You can just write `string source((std::istreambuf_iterator<char>(input_file)), std::istreambuf_iterator<char>());`  and this will read the whole file into the string directly